src/cuda/gpu_common.h

```diff
@@ -130,15 +134,14 @@ cudaEventDestroy(end);
 }
 
 
-
-
 // CUDA safe call
+#ifdef DEBUG
 #define QUICK_SAFE_CALL(x)\
 {\
 TIMERSTART()\
-printf("%s.%s.%d: %s\n", __FILE__, __FUNCTION__, __LINE__, #x); fflush(stdout);\
-printf("LAUCHBOUND = %i %i\n", gpu->blocks, gpu->twoEThreadsPerBlock);\
-printf("METHOD = %i\n", gpu->gpu_sim.method);\
+fprintf(gpu->debugFile,"%s.%s.%d: %s\n", __FILE__, __FUNCTION__, __LINE__, #x);fflush(gpu->debugFile);\
+fprintf(gpu->debugFile,"LAUCHBOUND = %i %i\n", gpu->blocks, gpu->twoEThreadsPerBlock);\
+fprintf(gpu->debugFile,"METHOD = %i\n", gpu->gpu_sim.method);\
 x;\
 TIMERSTOP()\
 cudaError_t error = cudaGetLastError();\
```

Comment:
I think we should keep error checking also in the production code. And perhaps the timer information although I don't know if it is used.

Reply:
Lets create a new project for QUICK error handling.